AsyncKeyManipulationSample/app/src/main/java/com/rivetz/rivetsample5/MainActivity.java

```diff
@@ -89,10 +94,32 @@ protected void onCreate(Bundle savedInstanceState) {
         });
     }
 
+    /**
+     * Called when getDeviceProperties completes
+     *
+     * @param result String result of the properties call
+     * @param throwable if not null, the exception
+     */
+    private void getPropertyComplete(String result, Throwable throwable) {
+        if (throwable == null) {
+            drtSupported = result;
+            if (result.equals("true"))
+                runOnUiThread(() -> alert("Paired, DRT supported"));
+            else
+                runOnUiThread(() -> alert("Paired, DRT not supported"));
+            makeClickable(findViewById(R.id.createKey));
+
+        } else {
+            runOnUiThread(() -> alert("Failed to get DRT properties"));
+        }
+    }
+
     public void onDevicePairing(boolean success){
         if (success) {
-            alert("Paired");
-            makeClickable(findViewById(R.id.createKey));
+            // Check if DRT is supported
+            RivetAndroid rivet = (RivetAndroid)crypto;
```

Comment:
Never use the RivetAndroid class directly because it is an implementation that may change. The reason you didn't find getDeviceProperty in the API is because it is implemented in RivetWalletActivity - that your activity extended. Please just call the method on your own class.

Change all of the samples the same way.